gbdxtools/images/landsat_image.py

```diff
@@ -48,6 +48,10 @@ def _ndwi_bands(self):
     def _build_graph(cls, _id, band_type="MS", proj=None, **kwargs):
         spec = band_types[band_type]
         landsat = rda.LandsatRead(landsatId=_id, productSpec=spec)
+        if kwargs.get('pansharpen') == True:
+            if spec == 'multispectral':
```

Comment:
Does this mean that providing `band_type="pan"` (or whatever the right term to specify panchromatic) along with `pansharpen=True` would _not_ return pansharpened, whereas `band_type="MS"` and `pansharpen=True` does return pansharpened? I understand that we make the band_type multispectral by default, so that when somebody simply passes in the pansharpen keyword, they'll get the correct result back, which makes sense right? If I say `im=ImageClass(id, pansharpen=True)` I would expect to get a pansharpened image. But if here I do the same but `im=ImageClass(id, band_type="pan", pansharpen=True)`, it would seem consistent to also return pansharpened here even though it means we're implementing a weird interface to image creation that implicitly makes override decisions on your input arguments according to some rules behind the scenes. Anyway, I think this is something we can make better and more consistent across our image classes, we ought to talk about in a new issue, I think there's some good stuff we can do.

Reply:
The thinking is that since pansharpening is the application of the panchromatic band to the visible imagery, it doesn't make sense to add the pan band to itself.

Reply:
ah, I tend to think of pansharpening as a function that takes (high-spatial-res/low-spectral-res, low-spatial-res/high-spectral-res) and ouputs image with high spectral and spatial res components

Reply:
so less of an application of one to the other since each provides the resolution source to make the pansharpened result

Reply:
I'm happy either way.

Reply:
Though what if `band_type=thermal`?

I also wonder if you're asking for `band_type=pan`, you're implying you're expecting only a single band to come back.

Reply:
yeah right this is kind of what I was alluding to earlier, it's a confusing interface that I think is the product of poor choice of semantics as well as the fact that keyword arguments were kind of developed to reflect specific rda operations... i think it'd be better if we forced band_type arguments to be full names, eg "panchromatic", "multispectral" etc.  part of me wants to just include "pansharpened" in supported band_types, but that abstracts the rda operator spec pattern

Reply:
Agree on the poor choise of semantics here 100%. I like the idea of only supporting band_type and having "pansharpened" or "pansharp" be a valid option.

Reply:
So its really a matter of what carries more weight, correct? either `band_type` or `pansharpen` im not sure what the answer is

Reply:
Im leaning towards placing more value on `pansharpen=True` always returning a pansharp image and ignoring band_type for now. Then opening an issue to completely overhaul the keyword pattern for pansharpening in favor of having "pansharp" be a valid option for band type. 

How does that sound @rjpolackwich and @drwelby ?

Reply:
@chelm yup for sure, that precedence is consistent with what we've always done, so otherwise would be introducing diff behavior without visible api change to users